### ISSUE TITLE
Looping a handler function in onChange when using map function.

### ISSUE THREAD

**Issue by corsicanec82** [sindresorhus/on-change#60]
Hello!

Please tell me why there is endless execution if I use the immutable map function. She does not change the object of state. Is this a bug or is there an explanation for this? If I remove the map function, then the handler is called once, as it should be. Here is the code that demonstrates this behavior:

```js
const onChange = require('on-change');

const object = {
  arr: [],
  foo: true,
};

const watchedObject = onChange(object, function (path) {
  console.log(path);
  // the call of the immutable map function
  // leads to an endless execution of the handler function
  this.arr.map((el) => {
  });
});

watchedObject.arr.unshift('value');

// => 'arr'
// => 'arr'
// => 'arr'
// => 'arr'
...
it happens endlessly
```

**Comment by DarrenPaulWright** [comment]
While map is immutable on the array itself, the elements in the array could be mutated within the callback, so we need to watch it. That said, I have a fix for the infinite loop that I will publish after I address some other issues.

**Comment by DarrenPaulWright** [comment]
Fixed in 2.0.2
